recipe/meta.yaml

```
@@ -0,0 +1,135 @@
+{% set version = "2.3.2" %}
+# Obtain from ZLIB_VERSION in zlib.h.in; https://github.com/zlib-ng/zlib-ng/blob/VERSION/zlib.h.in#L61
+{% set zlib_api = "1.3.1" %}
+# Please do not reset the build number unless zlib_api changed. It needs to be incremental across
+# both package versions.
+{% set number = 1 %}
+
+package:
+  name: zlib-ng
+  version: {{ version }}
+
+source:
+  url: https://github.com/zlib-ng/zlib-ng/archive/refs/tags/{{ version }}.tar.gz
+  sha256: 6a0561b50b8f5f6434a6a9e667a67026f2b2064a1ffa959c6b2dae320161c2a8
+
+build:
+  number: {{ number }}
+  script_env:
+    - ZLIB_COMPAT=0  # [compat == "no"]
+    - ZLIB_COMPAT=1  # [compat == "yes"]
+  run_exports:
+    # Just guessing since the library is so new
+    # but they seem to want to be backward compatible
+    # https://abi-laboratory.pro/index.php?view=timeline&l=zlib-ng
+    - {{ pin_subpackage('zlib-ng', max_pin='x.x') }}  # [compat == "no"]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ stdlib('c') }}
+    - cmake
+    - make  # [unix]
+    - gtest  # [compat == "yes"]
+
+test:
+  commands:
+    # compat builds a standalone z library that does not depend on zlib-ng
+    - test -e $PREFIX/include/zlib-ng.h                         # [unix and compat == "no"]
+    - test -e $PREFIX/include/zconf-ng.h                        # [unix and compat == "no"]
+    - test -e $PREFIX/lib/libz-ng${SHLIB_EXT}                   # [unix and compat == "no"]
+    - test -e $PREFIX/lib/pkgconfig/zlib-ng.pc                  # [unix and compat == "no"]
+    - if not exist %LIBRARY_INC%\zlib-ng.h exit 1               # [win and compat == "no"]
+    - if not exist %LIBRARY_INC%\zconf-ng.h exit 1              # [win and compat == "no"]
+    - if not exist %LIBRARY_BIN%\zlib-ng2.dll exit 1            # [win and compat == "no"]
+    - if not exist %LIBRARY_LIB%\zlib-ng.lib exit 1.            # [win and compat == "no"]
+    - if not exist %LIBRARY_LIB%\pkgconfig\zlib-ng.pc exit 1    # [win and compat == "no"]
+
+outputs:
+  # zlib-ng gets built when compat is NOT enabled; versioned with the regular zlib-ng release
+  - name: zlib-ng
+    build:
+      skip: true  # [compat == "yes"]
+  # zlib and libzlib get built when compat is enabled; they are versioned after the zlib API
+  - name: libzlib
+    version: {{ zlib_api }}
+    build:
+      skip: true  # [compat == "no"]
+      string: h{{ PKG_HASH }}_ng_{{ version.replace(".", "") }}_{{ number }}
```

If we make `(lib)zlib` depend on `zlib-ng-compat={version}`, then we don't need to add the version in the build string:

```suggestion
      string: h{{ PKG_HASH }}_ng_{{ number }}
```

---

Don't we want to keep it for readability, though?

---

No strong opinion, myself. I'll let you pick your fav!